File: go/src/github.com/eliudarudo/go-cli-app/player/utility.go
package player

import (
	"fmt"
	"strings"
)

// PrintWithBuffer function
func PrintWithBuffer(str string, bufferLength int) string {

	runes := []rune(str)
	/*
	   Using string builder most effective way to append
	   a character (rune) to a string
	*/
	var buffer strings.Builder
	for i := 0; i < bufferLength; i++ {
		if i < len(runes) {
			buffer.WriteRune(runes[i])
		} else {
			buffer.WriteString(" ")
		}
	}

	return buffer.String()
}

// StringIsValid function
func StringIsValid(str string) bool {
	if len(str) == 0 {
		return false
	}
	return true
}

// PrintFieldErrorMessage function
func PrintFieldErrorMessage() {
	fmt.Println("Please try again, invalid input")
}

File: server/src/lib/java/src/com/eliudarudo/cliapp/player/Utility.java
package com.eliudarudo.cliapp.player;

import java.lang.*;

public class Utility {
    public static String printWithBuffer(String string, int bufferLength) {
        StringBuilder buffer = new StringBuilder();
        for(int i=0; i < bufferLength; i++) {
            if(i < string.length()) {
                buffer.append(string.charAt(i));
            } else {
                buffer.append(" ");
            }
        }
        return buffer.toString();
    }

    public static boolean stringIsValid(String string) {
        try {
            return string.length() != 0;
        } catch(Exception e) {
            return false;
        }
    }

    public static void printFieldErrorMessage() {
        System.out.println("Please try again, invalid input");
    }
}
